### PR TITLE
feat: Automated testing, the request is proxied to the edge cluster regardless of whether it is an internal address or not

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Automated testing, the request is proxied to the edge cluster regardless of whether it is an internal address or not

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Automated testing, the request is proxied to the edge cluster regardless of whether it is an internal address or not         |
| 🇨🇳 中文    |       通过自动测试，无论请求是否为内部地址，都会将其代理到边缘群集       |

### CODE REVIEW

File: pkg/apitestsv2/netportal.go [hunk 1]
```diff
@@ -67,10 +67,10 @@ func useNetportal(inputURL string, netportalOpt *netportalOption) bool {
 	hostport := r.Host
 	ss := strings.SplitN(hostport, ":", 2)
 	host := ss[0]
-	// doesn't have k8s svc suffix, do not use nerportal
-	if !strings.HasSuffix(host, k8sServiceSuffix) {
-		return false
-	}
+	//// doesn't have k8s svc suffix, do not use nerportal
```
Comment (sfwn) [inline pkg/apitestsv2/netportal.go:70]: test if you can access default k8s namespace

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by kakj-go** [comment]
test image:
image

The address access URL will be changed

File: pkg/apitestsv2/netportal.go (hunk 1 above)
Comment (kakj-go) [reply pkg/apitestsv2/netportal.go:73]: image

The blacklist address of the edge cluster will not be accessed, but the central cluster will be accessed

**Comment by kakj-go** [comment]
Blacklist test:
image

File: pkg/apitestsv2/netportal.go [hunk 2]
```diff
@@ -52,32 +55,32 @@ func handleCustomNetportalRequest(apiReq *apistructs.APIRequestInfo, netportalOp
 }
 
 // useNetportal return true or false to represent use netportal or not.
-func useNetportal(inputURL string, netportalOpt *netportalOption) bool {
+func useNetportal(inputURL string, netportalOpt *netportalOption) (bool, error) {
```
Comment (sfwn) [inline pkg/apitestsv2/netportal.go:58]: if you must visit through netportal, this method is useless anymore.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/apitestsv2/apitest_test.go [hunk 3]
```diff
@@ -67,17 +67,9 @@ func TestAPITest_Invoke(t *testing.T) {
...
-				Headers: map[string][]string{"Accept-Encoding": {"identity"}},
-				Params:  map[string][]string{},
-				Body: apistructs.APIBody{
-					Content: "",
-				},
-			},
+			want:    nil,
 			want1:   &apistructs.APIResp{},
-			wantErr: false,
+			wantErr: true,
```
Comment (sfwn) [inline pkg/apitestsv2/apitest_test.go:72]: why err

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/apitestsv2/apitest_test.go (hunk 3 above)
Comment (kakj-go) [reply pkg/apitestsv2/apitest_test.go:72]: This one side will send a request, which cannot test whether the startup agent can send successfully, so only the err case is tested

**Review by sfwn — APPROVED** [review]

**Pull request merged**